File: app/src/main/java/com/icesoft/msdb/android/model/EventEdition.java
package com.icesoft.msdb.android.model;

import java.time.LocalDate;
import java.util.Calendar;
import java.util.Date;

import lombok.Data;

@Data
public class EventEdition extends BaseModel {

    private Long id;
    private Event event;
    private Integer[] eventDate;
    private RacetrackLayout trackLayout;
    private String posterUrl;
    private Boolean multidriver;
    private String location;

    public LocalDate getEventDate() {
        return LocalDate.of(eventDate[0] , eventDate[1], eventDate[2]);
    }

    public boolean isRally() {
        return event.isRally();
    }

    public boolean isRaid() {
        return event.isRaid();
    }
}

File: app/src/main/java/com/icesoft/msdb/android/model/EventEdition.kt
package com.icesoft.msdb.android.model

import java.time.LocalDate

data class EventEdition(
    val id: Long? = null,
    val longEventName: String? = null,
    val event: Event? = null,
    private val eventDate: Array<Int?>?,
    val trackLayout: RacetrackLayout? = null,
    val posterUrl: String? = null,
    val multidriver: Boolean? = false,
    val location: String? = null,
    val status: String? = null
): BaseModel() {

    constructor(): this(null, null,null, null, null, null, false, null, null)

    fun getEventDate(): LocalDate? {
        return LocalDate.of(
            eventDate!![0]!!,
            eventDate[1]!!,
            eventDate[2]!!
        )
    }

    fun isRally(): Boolean {
        return event!!.isRally()
    }

    fun isRaid(): Boolean {
        return event!!.isRaid()
    }
}
